libraries/lib-wave-track/WaveTrack.cpp

```diff
@@ -3930,6 +4071,58 @@ bool WaveTrack::MergeOneClipPair(int clipidx1, int clipidx2)
    return true;
 }
 
+void WaveTrack::ApplyStretchRatioOnIntervals(
+   const std::vector<IntervalHolder>& srcIntervals,
+   const ProgressReporter& reportProgress)
+{
+   std::vector<IntervalHolder> dstIntervals;
+   dstIntervals.reserve(srcIntervals.size());
+   std::transform(
+      srcIntervals.begin(), srcIntervals.end(),
+      std::back_inserter(dstIntervals), [&](const IntervalHolder& interval) {
+         return interval->GetStretchRenderedCopy(
+            reportProgress, *this, mpFactory);
+      });
+
+   // If we reach this point it means that no error was thrown - we can replace
+   // the source with the destination intervals.
+   for (auto i = 0; i < srcIntervals.size(); ++i)
+      ReplaceInterval(srcIntervals[i], dstIntervals[i]);
+}
+
+void WaveTrack::InsertInterval(const IntervalHolder& interval)
+{
```

Comment:
The assertions in this and the next two functions should be stated in header comments as preconditions.